devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,667 @@
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+metadata {
+	definition (name: "HELTUN TPS05 Switch", namespace: "HELTUN", author: "Sarkis Kabrailian", cstHandler: true, mcdSync: true ) {
+		capability "Temperature Measurement"
+		capability "Illuminance Measurement"
+		capability "Relative Humidity Measurement"
+		capability "Energy Meter"
+		capability "Power Meter"
+		capability "Configuration"
+		capability "Health Check"
+		capability "Refresh"
+
+		fingerprint mfr: "0344", prod: "0004", model: "0003", deviceJoinName: "HELTUN"
+	}
+	preferences {
+		input (
+			title: "HE-TPS05 | HELTUN Touch Panel Switch",
+			description: "The user manual document with all technical information is available in support.heltun.com page. In case of technical questions please contact HELTUN Support Team at support@heltun.com",
+			type: "paragraph",
+			element: "paragraph"
+		)
+			parameterMap().each {
+			if (it.title != null) {
+			input (
+					title: "${it.title}",
+					description: it.description,
+					type: "paragraph",
+					element: "paragraph"
+				)
+			}
+			def unit = it.unit ? it.unit : ""
+			def defV = it.default as Integer
+			def defVDescr = it.options ? it.options.get(defV) : "${defV}${unit} - Default Value"
+			input (
+				name: it.name,
+				title: null,
+				description: "$defVDescr",
+				type: it.type,
+				options: it.options,
+				range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+				defaultValue: it.default,
+				required: false
+			)
+		}
+	}
+}
+
+def checkParam() {
+	boolean needConfig = false
+	parameterMap().each {
+		if (state."$it.name" == null || state."$it.name".state == "defNotConfigured") {
+			state."$it.name" = [value: it.default as Integer, state: "defNotConfigured"]
+			needConfig = true
+		}
+		if (settings."$it.name" != null && (state."$it.name".value != settings."$it.name" as Integer || state."$it.name".state == "notConfigured")) {
+			state."$it.name".value = settings."$it.name" as Integer
+			state."$it.name".state = "notConfigured"
+			needConfig = true
+		}
+	}
+	if ( needConfig ) {
+		configParam()
+	}
+}
+
+private configParam() {
+	def cmds = []
+	for (parameter in parameterMap()) {
+	if ( state."$parameter.name"?.value != null && state."$parameter.name"?.state in ["notConfigured", "defNotConfigured"] ) {
+			cmds << zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$parameter.name".value, parameterNumber: parameter.paramNum, size: parameter.size).format()
+			cmds << zwave.configurationV2.configurationGet(parameterNumber: parameter.paramNum).format()
+			break
+		}
+	}
+	if (cmds) {
+		runIn(5, "checkParam")
+		sendHubCommand(cmds,500)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [:]
+	def roomTemperature = 1
+	def himidity = 5
```

Comment:
nit: `humidity` (and 1 more appearance)

Reply:
I didn't get what you mean, may I ask you to explain

Reply:
You've made a grammatical mistake here. `def himidity = 5`, should be `humidity`.